src/mixins/raster-layer-poly-mixin.js

```diff
@@ -644,13 +649,34 @@ export default function rasterLayerPolyMixin(_layer) {
       })
     }
 
+    // Contour polys may be colored by contour value rather than a color column
+    if (contourAutocolors) {
```

Comment:
I think it would be preferable to roll this auto color logic into the if/else above `if(isContourType(state)) { ... } else { ... }` and use the same "auto" value if possible. That way we could avoid creating a special value for contour chart in `auto-contour`

Reply:
`auto` is used in the logic to pull scales for legends pretty removed from this context; I’d rather not add more exceptions/pass extra layer type context around if there is no overlapping logic with `auto`. If/when contour supports legends, sure

Reply:
Makes sense but I wonder if there's another way to disable the legend without a special value, or if the decision could be made above this where a reference to chart is available (`toLegendState` or something). Either way, works for now until legends are a thing